dp_and_yang_push/route-status.cpp

```diff
@@ -0,0 +1,575 @@
+        confd_fatal("Failed to confd_connect() to confd \n");
+
+    if (confd_register_trans_cb(dctx, &trans) == CONFD_ERR)
+        confd_fatal("Failed to register trans cb \n");
+
+    if (confd_register_data_cb(dctx, &data) == CONFD_ERR)
+        confd_fatal("Failed to register data cb \n");
+
+    struct confd_push_on_change_cbs pcb;
+    memset(&pcb, 0, sizeof(pcb));
+    pcb.fd = workersock;
+    pcb.subscribe_on_change = cb_subscribe_on_change;
+    pcb.unsubscribe_on_change = cb_unsubscribe_on_change;
+    pcb.cb_opaque = NULL;
+    strcpy(pcb.callpoint, rs__callpointid_routestat);
+    if (confd_register_push_on_change(dctx, &pcb) == CONFD_ERR)
+        confd_fatal("Failed to register push on change cb \n");
+
+    if (confd_register_done(dctx) != CONFD_OK)
+        confd_fatal("Failed to complete registration \n");
+    INFO_EXIT("");
+}
+
+void generate_changes()
+{
+    INFO_ENTER("");
+    TRACE("Num of routes=%i", routes.size());
+    while (true) {
+        int rt_idx = rand() % routes.size();
+        TRACE("tr_idx=%i", rt_idx);
+        route_type *rt = NULL;
+        rt = routes[rt_idx].get();
+        rt->leaf1 = rand() % 10 + 1;
+        cb_push_on_change(rt);
+        sleep(2);
+    }
+    INFO_EXIT("");
+}
+
+int confd_loop()
+{
+    INFO_ENTER("");
+    int ret = CONFD_OK;
+
+    while (1) {
+        struct pollfd set[2];
+
+        set[0].fd = ctlsock;
+        set[0].events = POLLIN;
+        set[0].revents = 0;
+
+        set[1].fd = workersock;
+        set[1].events = POLLIN;
+        set[1].revents = 0;
+
+        if (poll(set, sizeof(set) / sizeof(*set), -1) < 0) {
+            perror("Poll failed:");
+            break;
+        }
+
+        /* Check for I/O */
+        if (set[0].revents & POLLIN) {
+            if ((ret = confd_fd_ready(dctx, ctlsock)) == CONFD_EOF) {
+                confd_fatal("Control socket closed\n");
+            } else if (ret == CONFD_ERR && confd_errno != CONFD_ERR_EXTERNAL) {
+                confd_fatal("Error on control socket request: %s (%d): %s\n",
+                        confd_strerror(confd_errno), confd_errno,
+                        confd_lasterr());
+            }
+        }
+        if (set[1].revents & POLLIN) {
+            if ((ret = confd_fd_ready(dctx, workersock)) == CONFD_EOF) {
+                confd_fatal("Worker socket closed\n");
+            } else if (ret == CONFD_ERR && confd_errno != CONFD_ERR_EXTERNAL) {
+                confd_fatal("Error on worker socket request: %s (%d): %s\n",
+                        confd_strerror(confd_errno), confd_errno,
+                        confd_lasterr());
+            }
+        }
+    }
+
+    INFO_EXIT("ret %i", ret);
+    return ret;
+}
+
+// pass one numeric parameter specifying nunmber of records (eg. ./route-status 1000)
+int main(int argc, char *argv[])
+{
+    INFO_ENTER("argc %i", argc);
+    int ret = CONFD_OK;
+    bool arg_ok = true;
+
+    if (argc != 2) {
+        arg_ok = false;
+    }
+    try {
```

Comment:
bit broken - arg_ok could be already false at this point

Reply:
will fix